Userland/Libraries/LibELF/DynamicLoader.cpp

```diff
@@ -803,8 +803,8 @@ void DynamicLoader::call_object_init_functions()
     typedef void (*InitFunc)();
 
     if (m_dynamic_object->has_init_section()) {
-        auto init_function = m_dynamic_object->init_section_function();
-        (init_function)();
+        dbgln("DynamicLoader: Object has obsolete init_section {}", filepath());
+        VERIFY_NOT_REACHED();
     }
```

Comment:
I would also check for DT_FINI somewhere. We currently call it here on exit: https://github.com/SerenityOS/serenity/blob/350fc077120fa4689c6f377f1d7c6f0afa0a4e56/Userland/Libraries/LibELF/DynamicLinker.cpp#L614-L617

Since we don't need the actual DT_INIT/DT_FINI address, we could maybe remove all associated functions/variables and throw the errors here?:
https://github.com/SerenityOS/serenity/blob/350fc077120fa4689c6f377f1d7c6f0afa0a4e56/Userland/Libraries/LibELF/DynamicObject.cpp#L82-L87

Reply:
Done!